semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala

```diff
@@ -25,44 +26,64 @@ object Main {
+          if (PathIO.extension(file) != "class") return FileVisitResult.CONTINUE
+          try {
+            val relpath = AbsolutePath(file).toRelative(root).toString
+            val bytecode = ByteCode(Files.readAllBytes(file))
+            val classfile = ClassFileParser.parse(bytecode)
+            ScalaSigParser.parse(classfile) match {
+              case Some(scalaSig) =>
+                val semanticdbInfos = scalaSig.symbols.flatMap {
+                  case sym: SymbolInfoSymbol => sinfo(sym)
+                  case _ => None
+                }
+                val className = NameTransformer.decode(PathIO.toUnix(relpath))
+                val semanticdbRelpath = relpath + ".semanticdb"
+                val semanticdbAbspath = semanticdbRoot.resolve(semanticdbRelpath)
+                val semanticdbDocument = s.TextDocument(
+                  schema = s.Schema.SEMANTICDB3,
+                  uri = className,
+                  language = "Scala",
+                  symbols = semanticdbInfos)
+                val semanticdbDocuments = s.TextDocuments(List(semanticdbDocument))
+                FileIO.write(semanticdbAbspath, semanticdbDocuments)
+              case None =>
+                // NOTE: If a classfile doesn't contain ScalaSignature,
+                // we skip it for the time being. In the future, we may add support
+                // for parsing arbitrary Java classfiles.
+                ()
             }
-            val className = NameTransformer.decode(PathIO.toUnix(fragment.name.toString))
-            val semanticdbRelpath = fragment.name.resolveSibling(_ + ".semanticdb")
-            val semanticdbAbspath = semanticdbRoot.resolve(semanticdbRelpath)
-            val semanticdbDocument = s.TextDocument(
-              schema = s.Schema.SEMANTICDB3,
-              uri = className,
-              language = "Scala",
-              symbols = semanticdbInfos)
-            val semanticdbDocuments = s.TextDocuments(List(semanticdbDocument))
-            FileIO.write(semanticdbAbspath, semanticdbDocuments)
-          case None =>
-            // NOTE: If a classfile doesn't contain ScalaSignature,
-            // we skip it for the time being. In the future, we may add support
-            // for parsing arbitrary Java classfiles.
-            ()
+          } catch {
+            case NonFatal(ex) =>
+              println(s"error: can't convert $file")
+              ex.printStackTrace()
+              failed = true
+          }
+          FileVisitResult.CONTINUE
         }
-      } catch {
-        case NonFatal(ex) =>
-          println(s"error: can't convert $fragment")
-          ex.printStackTrace()
-          failed = true
+
+        override def postVisitDirectory(dir: Path, e: IOException): FileVisitResult = {
+          if (e != null) throw e
+          FileVisitResult.CONTINUE
+        }
+
+        override def visitFileFailed(file: Path, e: IOException): FileVisitResult = throw e
```

Comment:
What about exceptions that we get here? Shall we also catch and println them without crashing?